core/services/ardupilot_manager/mavlink_proxy/AbstractRouter.py

```diff
@@ -112,11 +112,11 @@ async def exit(self) -> None:
             if self._subprocess is not None:
                 logger.error("terminate")
                 self._subprocess.terminate()
-                await asyncio.sleep(1)  # Non-blocking sleep
+                await asyncio.sleep(3)  # Non-blocking sleep
                 if await self.is_running():
                     logger.error("kill")
                     self._subprocess.kill()
-                    await asyncio.sleep(1)
+                    await asyncio.sleep(3)
```

Comment:
can't we await for the kill to return instead of waiting for a fixed time?

Reply:
perhaps:
```python
self._subprocess.kill()
await self._subprecess.wait()
```

Reply:
for that we could move to asyncio subprocess, we could work that on a future PR and open an issue.

Reply:
ouch, you are correct, will update

Reply:
@joaoantoniocardoso btw, we already such thing on the following line

Reply:
uh, I wonder why that isn't enough?

Reply:
I'll do more tests to be sure

Reply:
1. if it is running, terminate the process
2. wait for 3 seconds
3. if it is still running, kill the process
4. wait for 3 seconds (why, what for?)
5. wait for the process

Reply:
maybe something using a timeout like:

```python
if await self.is_running():
  if self._subprocess is not None:

    self._subprocess.terminate()
    try:
      await asyncio.wait_for(self._subprocess, timeout=3)
    except asyncio.TimeoutError:
      if await self.is_running(): 
        self._subprocess.kill()
        await self._subprecess.wait() # Wait for the subprocess to terminate
        # or maybe instead we need to: await self._subprocess.communicate()
```

Reply:
I did try that and now the issue is back again. Since we have limited time, is there a problem with the original approach ? Besides the verbosity/timed logic.

Reply:
> I did try that and now the issue is back again. Since we have limited time, is there a problem with the original approach ? Besides the verbosity/timed logic.

The original approach seems fragile. It might not work on different binary versions or other scenarios like different CPU loads...